src/components/GasFeeWarning.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet } from 'react-native'
+import AppAnalytics from 'src/analytics/AppAnalytics'
+import { AppEvents } from 'src/analytics/Events'
+import InLineNotification, { NotificationVariant } from 'src/components/InLineNotification'
+import { CICOFlow } from 'src/fiatExchanges/types'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { Spacing } from 'src/styles/styles'
+import { PreparedTransactionsResult } from 'src/viem/prepareTransactions'
+
+export type GasFeeWarningFlow = 'Send' | 'Swap' | 'Withdraw' | 'Deposit' | 'Dapp'
+
+function GasFeeWarning({
+  prepareTransactionsResult,
+  flow,
+  changeInputValueFn,
+  testIdPrefix,
+}: {
+  prepareTransactionsResult?: PreparedTransactionsResult
+  flow: GasFeeWarningFlow
+  changeInputValueFn?: (amount: string) => void
+  testIdPrefix?: string
+}) {
+  const { t } = useTranslation()
+
+  useEffect(() => {
+    if (prepareTransactionsResult && prepareTransactionsResult.type !== 'possible') {
+      AppAnalytics.track(AppEvents.gas_fee_warning_impression, {
+        flow,
+        errorType: prepareTransactionsResult.type,
+        tokenId: feeCurrency.tokenId,
+      })
+    }
+  }, [prepareTransactionsResult])
+
+  if (!prepareTransactionsResult || prepareTransactionsResult.type === 'possible') {
+    return false
+  }
+
+  const feeCurrency =
+    prepareTransactionsResult.type === 'not-enough-balance-for-gas'
+      ? prepareTransactionsResult.feeCurrencies[0]
+      : prepareTransactionsResult.feeCurrency
```

Comment:
i wonder if we'll ever add another error type/string to the transaction result object -- as it is now, we catch the `need-decrease-spend-amount-for-gas` case implicitly, but if we ever add anything else, this will break silently. it's probably not a big deal, but wonder if we should be more explicit about that check?